### ISSUE TITLE
Linux support to open links in the user's browser

### ISSUE THREAD

**Issue by mariopal** [SKCraft/Launcher#427]
Some buttons and texts in the Launcher are links to be opened in the user's web browser, but on Linux "Desktop.getDesktop().Browse(...)" is not well supported and such links are not opened.

One way to fix this is to use the following patch in the code, which I have tested successfully, but maybe there is some more standardized way to do it:
```diff
diff --git a/launcher/src/main/java/com/skcraft/launcher/swing/SwingHelper.java b/launcher/src/main/java/com/skcraft/launcher/swing/SwingHelper.java
index c06b6c0..478c1a1 100644
--- a/launcher/src/main/java/com/skcraft/launcher/swing/SwingHelper.java
+++ b/launcher/src/main/java/com/skcraft/launcher/swing/SwingHelper.java
@@ -103,6 +103,24 @@ public final class SwingHelper {
     public static void openURL(URL url, Component parentComponent) {
         try {
             Desktop.getDesktop().browse(url.toURI());
+        } catch (UnsupportedOperationException eek) {
+            String myOS = System.getProperty("os.name").toLowerCase();
+            Runtime runtime = Runtime.getRuntime();
+
+            if (myOS.contains("mac")) { // macOS
+                try {
+                    runtime.exec("open " + url.toURI());
+                } catch (IOException | URISyntaxException e) {
+                    return;
+                }
+            }
+            else if (myOS.contains("nix") || myOS.contains("nux")) { // Linux
+                try {
+                    runtime.exec("xdg-open " + url.toURI());
+                } catch (IOException | URISyntaxException e) {
+                    return;
+                }
+            }
         } catch (IOException e) {
             showErrorDialog(parentComponent, tr("errors.openUrlError", url.toString()), SharedLocale.tr("errorTitle"));
         } catch (URISyntaxException e) {
```

**Comment by hedgehog1029** [comment]
I actually develop on and actively use Linux and recently implemented and tested the Microsoft authentication dance using the same method, so it's a bit odd that you're having issues with toolkit support. Lemme go re-read the actual implementation, though.

**Comment by hedgehog1029** [comment]
Right, so the Linux `XToolkit` only supports desktop actions if GTK is installed - since it uses `gnome_url_show`. Swing LAF and toolkit support on Linux generally requires GTK so this isn't super surprising. Curious about the Mac part of the example code, since I'd expect the Mac toolkit to generally "just work" via built-in OSX API.

**Comment by hedgehog1029** [comment]
Fixed in 94ae835
